Workflow/Sources/WorkflowAction.swift

```diff
@@ -27,14 +27,22 @@ public protocol WorkflowAction<WorkflowType> {
     ///
     /// - Returns: An optional output event for the workflow. If an output event is returned, it will be passed up
     ///            the workflow hierarchy to this workflow's parent.
-    func apply(toState state: inout WorkflowType.State) -> WorkflowType.Output?
+    func apply(
```

Comment:
Any thoughts around offering both the old and the new functions to avoid changes to lots of files?

I.e., if you added the new one, along with a default impl that called the old one, could we get away with only changing existing action types that need access to the `context`?

Reply:
this was [explored](https://github.com/square/workflow-swift/pull/259) a few times in the past and i believe the conclusion was that it would be better to 'rip the bandaid off' so to speak. IMO the primary problem with trying to be source compatible is that you end up having to expose sort of redundant protocol requirements. yes, you can default the one with the new param to forward to the one without it, but it's still sort of an odd API. i think ideally you'd want them to be mutually exclusive, but i'm not sure that's expressible.

Reply:
that being said – if you've got alternate ideas, happy to hear them!

Reply:
Jamie has already explored ways to automate a large part of this migration, so I am fine with making the breaking change. It's why we have semver!

Reply:
Yeah, that makes sense - thanks for sharing the previous exploration!